### ISSUE TITLE
Generic type parsing does not work properly

### ISSUE THREAD

**Issue by strawb3rryx7** [scribd/Weaver#137]
The following code produces an error although it has correct syntax.

```swift
public class AnyRepository<RealmObject: Object>: AnyRepositoryProtocol {
    @Weaver(.reference)
    private var realmProvider: RealmProvider
}
```

**Error**: Invalid token ':' in type 'AnyRepository<RealmObject: Object>: '.

**Comment by strawb3rryx7** [comment]
Simple workaround defining constraints like the below;

```swift
public class AnyRepository<RealmObject>: AnyRepositoryProtocol where RealmObject : Object {
```

**Comment by strawb3rryx7** [comment]
The workaround can be applied by defining constraints as follows.

```swift
public class AnyRepository<RealmObject>: AnyRepositoryProtocol where RealmObject : Object {
```

**Comment by trupin** [comment]
Fixed by #142
